server/sandbox_network.go

```diff
@@ -68,17 +67,13 @@ func (s *Server) networkStart(ctx context.Context, sb *sandbox.Sandbox) (podIPs
 		return nil, nil, fmt.Errorf("failed to get network JSON for pod sandbox %s(%s): %v", sb.Name(), sb.ID(), err)
 	}
 
-	for idx, podIPConfig := range network.IPs {
-		podIP := strings.Split(podIPConfig.Address.String(), "/")[0]
-
-		// Apply the hostport mappings only for the first IP to avoid allocating
-		// the same host port twice
-		if idx == 0 && len(sb.PortMappings()) > 0 {
-			ip := net.ParseIP(podIP)
-			if ip == nil {
-				return nil, nil, fmt.Errorf("failed to get valid ip address for sandbox %s(%s)", sb.Name(), sb.ID())
-			}
+	for _, podIPConfig := range network.IPs {
+		ip := podIPConfig.Address.IP
+		if ip.To16() == nil {
+			return nil, nil, fmt.Errorf("failed to get valid ip address for sandbox %s(%s)", sb.Name(), sb.ID())
```

Comment:
`To16` can only fail if the IP is neither IPv4 nor IPv6, in which case it should have failed some check long before this